rules-reviewed/openshift/session.windup.xml

```diff
@@ -72,29 +63,22 @@
             <perform>
                 <hint title="HTTP Session data storage" effort="3" category-id="cloud-optional">
                     <message>
-                    <![CDATA[
-                    The servlet container uses HttpSession to create a session between an HTTP client and an HTTP server.
-                    The session persists for a specified time period, across more than one connection or page request from the user.
-                    `HttpSession.setAttribute` method allows user information to persist across multiple user connections
-                    Warning: As of Java Servlet Version 2.2, the `javax.http.HTTPSession.putValue(java.lang.String name,java.lang.Object value)` method is deprecated and replaced by `javax.http.HttpSession.setAttribute(java.lang.String, java.lang.Object)`
+                    In a cloud environment, a restart can wipe out HTTP session data in the memory of a running container.
 
-                    In a cloud environment it has to be considered that the data in the memory of a running container can be wiped out by a restart (triggered by code deploy, config change, or the execution environment relocating the process to a different physical location)[1].
-                    Consider storing HTTPSession data to a cache backing service[2][3][4][5]
-                    This approach has some benefits:
+                    Recommendations:
 
-                    * Increased application scalability and elasticity. By offloading the session data off to a remote Data Grid, the application tier itself can be more scalable and elastic.
-                    * Session Persistence. By offloading session data to a remote data grid, the application itself will be able to survive EAP node failures since the a JVM failure will not cause the session data to be lost.
-                    * Session Data Sharing. If you have a requirement for multiple applications to be able to share session data, this solution might be able to solve that use case as well.
+                    Store HTTP session data in a cache backing service or a remote data grid.
```

Comment:
in this case there's only 1 recommendation .... does it still make sense to have the heading "Recommentations:" ?

Reply:
It's like "Prerequisites". We use that title even if there is only ONE prerequisite. But doing that in a ruleset might be overkill. I'll check.